Classes/Action/SubscribeAction.php

```diff
@@ -210,6 +212,15 @@ protected function handleForm()
 
         $handler->handleForm();
 
+        $categoryIds = $this->getConfigurations()->getExploded('categories');
```

Comment:
* move to handle categories method
* dont destroy the context and insert code between the subscriber handling if possible